### PR TITLE
Updating to Infinispan 15.0.4

### PR DESCRIPTION
**Pull request by wburns**
Updates usage of Infinispan for both clustering (controller and agents) as well as the hotrod module to the latest stable version (15.0.4).

### CODE REVIEW

**Comment by wburns** [comment]
This relies upon https://github.com/Hyperfoil/Hyperfoil/pull/331 to build as Infinispan 15 requires Java 17 now.

**Comment by jabolina** [comment]
Got a failure locally. On CLI:

```
in-vm: null: Cannot instantiate class 'org.wildfly.security.sasl.plain.WildFlyElytronSaslPlainProvider'
```

The complete stack trace is:

```java
20:37:50,254 ERROR (vert.x-eventloop-thread-2) [i.h.c.i.SimulationRunner] One of the HTTP client pools failed to start. java.lang.ExceptionInInitializerError
	at org.infinispan.client.hotrod.impl.transport.netty.ChannelFactory.createChannelInitializer(ChannelFactory.java:215)
	at org.infinispan.client.hotrod.impl.transport.netty.ChannelFactory.newPool(ChannelFactory.java:207)
	at java.base/java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1708)
	at org.infinispan.client.hotrod.impl.transport.netty.ChannelFactory.fetchChannelAndInvoke(ChannelFactory.java:326)
	at org.infinispan.client.hotrod.impl.transport.netty.ChannelFactory.pingServersIgnoreException(ChannelFactory.java:239)
	at org.infinispan.client.hotrod.impl.transport.netty.ChannelFactory.start(ChannelFactory.java:172)
	at org.infinispan.client.hotrod.RemoteCacheManager.actualStart(RemoteCacheManager.java:359)
	at org.infinispan.client.hotrod.RemoteCacheManager.start(RemoteCacheManager.java:315)
	at org.infinispan.client.hotrod.RemoteCacheManager.<init>(RemoteCacheManager.java:174)
	at org.infinispan.client.hotrod.RemoteCacheManager.<init>(RemoteCacheManager.java:131)
	at io.hyperfoil.hotrod.connection.HotRodRemoteCachePoolImpl.start(HotRodRemoteCachePoolImpl.java:49)
	at io.hyperfoil.hotrod.HotRodRunData.lambda$openConnections$0(HotRodRunData.java:58)
	at io.vertx.core.impl.ContextImpl.lambda$executeBlocking$0(ContextImpl.java:177)
	at io.vertx.core.impl.ContextInternal.dispatch(ContextInternal.java:276)
	at io.vertx.core.impl.ContextImpl.lambda$internalExecuteBlocking$2(ContextImpl.java:209)
	at io.vertx.core.impl.TaskQueue.run(TaskQueue.java:76)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: org.infinispan.commons.CacheConfigurationException: Cannot instantiate class 'org.wildfly.security.sasl.plain.WildFlyElytronSaslPlainProvider'
	at org.infinispan.commons.util.Util.loadClass(Util.java:139)
	at org.infinispan.commons.util.Util.getInstance(Util.java:312)
	at org.infinispan.client.hotrod.impl.transport.netty.ChannelInitializer.<clinit>(ChannelInitializer.java:77)
	... 20 more
	Suppressed: java.lang.ClassNotFoundException: org.wildfly.security.sasl.plain.WildFlyElytronSaslPlainProvider
		at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641)
		at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188)
		at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:526)
		at java.base/java.lang.Class.forName0(Native Method)
		at java.base/java.lang.Class.forName(Class.java:534)
		at java.base/java.lang.Class.forName(Class.java:513)
		at org.infinispan.commons.util.Util.loadClassStrict(Util.java:179)
		at org.infinispan.commons.util.Util.loadClass(Util.java:137)
		... 22 more
```

File: hotrod/pom.xml [hunk 1]
```diff
@@ -275,7 +207,7 @@
                             <skip>${module.skipCopyDependencies}</skip>
                             <includeScope>runtime</includeScope>
                             <excludeScope>test</excludeScope>
-                            <excludeGroupIds>com.github.ben-manes.caffeine,org.infinispan,org.infinispan.protostream,org.wildfly</excludeGroupIds>
+                            <excludeGroupIds>com.github.ben-manes.caffeine,org.infinispan.protostream,org.wildfly</excludeGroupIds>
```
Comment (jabolina) [inline hotrod/pom.xml:210]: Seems removing the `org.wildfly` from here does the trick.

**Comment by wburns** [comment]
Oh whoops I didn't push the update with the wildly portion taken out.

File: hotrod/pom.xml (hunk 1 above)
Comment (wburns) [reply hotrod/pom.xml]: Yeah, I missed the force push to update this before, all fixed now.

**Comment by jabolina** [comment]
Just to update it is working fine on the latest commit.

**Comment by lampajr** [comment]
IIUC this depends on https://github.com/Hyperfoil/Hyperfoil/pull/331, right? If so I'd say let's rebase that one and once merged we can rebase this one as well

**Comment by lampajr** [comment]
Ok https://github.com/Hyperfoil/Hyperfoil/pull/331 has been merged, we can rebase this one on top of `master`

**Comment by wburns** [comment]
Rebased!

**Review by lampajr — APPROVED** [review]
I am not that expert on this part but it looks good to me!

@franz1981 any thoughts?

**Comment by jabolina** [comment]
We've released 15.0.5.Final recently, it should be nice to go for that already. I think I can't push to Will's branch to update it :confused:. But either way, just going from 13 to 15 is already a great improvement.

On another note, I've been running the benchmarks locally from this branch, and it's working well.

**Comment by lampajr** [comment]
> We've released 15.0.5.Final recently, it should be nice to go for that already. I think I can't push to Will's branch to update it 😕. But either way, just going from 13 to 15 is already a great improvement.

yeah we can do another upgrade later on.

> On another note, I've been running the benchmarks locally from this branch, and it's working well.

That sounds a great confirmation, thanks!

**Review by franz1981 — APPROVED** [review]
Thanks @wburns ready to go!

**Comment by jabolina** [comment]
Thanks, @lampajr @franz1981!

**Comment by franz1981** [comment]
@jabolina I've noticed
```
/home/runner/work/Hyperfoil/Hyperfoil/clustering/src/main/java/io/hyperfoil/HyperfoilChannelLookup.java:[20,47] [removal] JGroupsChannelLookup in org.infinispan.remoting.transport.jgroups has been deprecated and marked for removal
```
warn in the Ci after this merge: is it something we should be worried about?

**Comment by jabolina** [comment]
It should be fine. I'll try to open a PR with an equivalent version to remove the warning this week or next.

**Pull request merged**